dadi/lib/controller/index.js

```diff
@@ -33,11 +34,44 @@ function ApiError (status, code, message, title) {
   return err
 }
 
-function prepareQueryOptions (options, modelSettings) {
+var Controller = function (model) {
+  if (!model) throw new Error('Model instance required')
+  this.model = model
+  this.searchHelper = new SearchHelper(model)
+}
+
+Controller.prototype.get = function (req, res, next) {
+  var path = url.parse(req.url, true)
+  var options = path.query
+
+  // determine if this is jsonp
+  var done = options.callback ? sendBackJSONP(options.callback, res, next) : sendBackJSON(200, res, next)
+
+  var query = this.prepareQuery(req)
+  var queryOptions = prepareQueryOptions(options)
+
+  if (queryOptions.errors.length !== 0) {
+    done = sendBackJSON(400, res, next)
+    return done(null, queryOptions)
+  } else {
+    queryOptions = queryOptions.queryOptions
+  }
+
+  this.model.get(query, queryOptions, done, req)
+}
+
+const prepareQueryOptions = function (options, modelSettings) {
   var response = { errors: [] }
   var queryOptions = {}
   var settings = modelSettings || {}
 
+  if (options.search) {
+    let searchQuery = parseQuery(options.search)
```

Comment:
The build is breaking on versions 4 and 5 of Node because there's a missing strict mode header in this file.